app/src/main/java/ch/epfl/polychef/pages/EntryPage.java

```diff
@@ -56,15 +63,62 @@ protected void onResume() {
     }
 
     protected void goHomeIfConnected() {
-        if(FirebaseAuth.getInstance().getCurrentUser() != null) {
-            startActivity(new Intent(this, HomePage.class));
+        if(getFireBaseAuth().getCurrentUser() != null) {
+            findViewById(R.id.autoLoginBackground).setVisibility(View.VISIBLE);
+            findViewById(R.id.autoLoginProgress).setVisibility(View.VISIBLE);
+
+            preventInteractions();
+
+            Toast.makeText(this, "Auto connect" , Toast.LENGTH_SHORT).show();
+
+            getUserStorage().initializeUserFromAuthenticatedUser(this);
         }
     }
 
+    private void preventInteractions(){
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
```

Comment:
That's cool, could you send me the link where you found this way of preventing interactions ?

Reply:
I think it was from [here](https://stackoverflow.com/questions/36918219/how-to-disable-user-interaction-while-progressbar-is-visible-in-android)

Reply:
thanks !